### DISCUSSION TITLE
Custom class and autoincrement

### DISCUSSION THREAD

**Discussion by CharlotteJu** [simolus3/drift#2689] [Q&A] [answered]
Hi,

I would like to use custom classes for my tables, so I'm using the @UseRowClass annotation. But how can I handle auto-incrementing IDs?

Let me explain with an example:

```
@UseRowClass(User)
class Users extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get name => text()();
  DateTimeColumn get birthday => dateTime()();
}

class User {
  final int id;
  final String name;
  final DateTime birthday;

  User({required this.id, required this.name, required this.birthday});

  String get birthdayString => DateFormat.ymd().format(birthday);
}
```

Normally, if I want to create a User with information gradually, I use the UserCompanion. However, in this case, if I want to use the getBirthdayString method, I need an object of type User, which requires a non-null ID. Do you have any ideas on how I should approach this?

P.S.: My classes have methods much more complex than getBirthdayString; it's just for illustration purposes.

**Comment by simolus3** [comment, marked as answer]
Since it's your class, couldn't you make `id` nullable in that case? Drift shouldn't complain about that, it can read non-null columns into nullable Dart files.

**Reply by CharlotteJu** [reply]
Yes ok, thanks ! I thought the types had to be exactly the same (nullable included).
